src/client.py

```diff
@@ -0,0 +1,362 @@
+                               'attribute', 'growth_form'],
+            'obshabitatcomplexities': ['habitatcomplexity', 'habitatcomplexity__transect',
+                                       'habitatcomplexity__transect__sample_event', 'score']
+        }
+        # Valid observation and filter for observation.
+        if obs not in obs_filters or filter not in obs_filters[obs]:
+            return None
+        # API observations path.
+        return self.api_projects(name=name, id=id, resource=obs, filter=filter, filter_val=filter_val)
+
+    def get_sample_unit(self, unit, filter=None, filter_val=None, name=None, id=None):
+        """
+        Gets sample unit.
+        :param unit: Sample unit name. Same units: fishbelttransects, benthictransects, quadratcollections. Visit
+        https://mermaid-api.readthedocs.io/en/latest/projects.html#sample-units for more information on sample units.
+        :type unit: str
+        :param filter: The only useful filters are likely to be len_surveyed_min/len_surveyed_max for
+         fishbelttransects and benthictransects.
+        :type filter: str
+        :param filter_val: (optional) Required for filters requiring values, eg.(len_surveyed_min/len_surveyed_max).
+        :type filter_val: int
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type id: str
+        :return: Same unit data.
+        :rtype: dict, None
+        """
+        unit_filters = ['len_surveyed_min', 'len_surveyed_max']
+        # Valid unit and filter exits and in valid unit filters.
+        if unit not in self.samples or filter and filter not in unit_filters:
+            return None
+        # API sample unit path.
+        return self.api_projects(name=name, id=id, resource=unit, filter=filter, filter_val=filter_val)
+
+    def get_sample_method(self, method, name=None, id=None):
+        """
+        Gets sample unit method.
+        :param method: Sample unit methods: beltfishtransectmethods, benthiclittransectmethods,
+        benthicpittransectmethods, habitatcomplexitytransectmethods, bleachingquadratcollectionmethods,
+        sampleunitmethods.
+        See https://mermaid-api.readthedocs.io/en/latest/projects.html#sample-unit-methods for more information.
+        :type method: str
+        :param name: (optional if 'id' provided) MERMAID project name.
+        :type name: str
+        :param id: (optional if 'name' provided) MERMAID project ID.
+        :type id: str
+        :return: Sample unit methods data.
+        :rtype: dict, None
+        """
+        if method not in self.samples:
+            return None
+        # API call sample unit methods path.
+        return self.api_projects(name=name, id=id, resource=method)
+
+    def get_sample_events(self, filter=None, filter_val=None, name=None, id=None):
```

Comment:
For these last five wrapper methods, you might change the order of the arguments to be more convenient for calls to it using positional args: `id=None, name=None, filter=None, filter_val=None`